tests/test_1.py

```diff
@@ -52,37 +52,32 @@
 }
 
 
-def castles_number(heights):
-    start_height, current_height = None, None
-    castles = 0
-
+def get_castles_number_1(heights):
     if len(heights) == 0:
-        raise Exception
+        raise Exception('Border is empty')
     if len(heights) == 1:
         return 1
 
+    castles = 0
+    current_height = heights[0]
+
     for i in range(len(heights)):
-        if current_height is None:
-            current_height = heights[i]
 
         if heights[i] != current_height:
-            if (start_height is None) or (heights[i] < current_height and start_height < current_height) or \
-                    (heights[i] > current_height and start_height > current_height):
+            if ('start_height' not in locals()
+                    or heights[i] < current_height and start_height < current_height
+                    or heights[i] > current_height and start_height > current_height):
                 castles += 1
-                start_height = heights[i - 1]
-                current_height = heights[i]
-            elif (heights[i] < current_height and start_height >= current_height) or \
-                    (heights[i] > current_height and start_height <= current_height):
-                start_height = heights[i - 1]
-                current_height = heights[i]
-
-        if heights[i] == current_height and i == len(heights) - 1:
+            start_height = heights[i - 1]
+            current_height = heights[i]
+
+        if i == len(heights) - 1:
```

Comment:
`if i == len(heights) - 1:`
What is the purpose/sense of this??
"if for loop is ending do this.." - is it the sense? I thought you know how to do it right.
```
for i in range(4):
   print('a')
else:
   print('b')
```

And when you will do it right I will ask you "Why do you need this logic? Can the `for` loop be finished before his end? Do you have `break`? No. It means this code will be preformed **always** at the end of the loop. So why you can't set `castles=1` instead this strange logic?"